automation/workflow/trigger.go

```diff
@@ -0,0 +1,23 @@
+package workflow
+
+import (
+// "encoding/json"
+)
+
+// TriggerType is the type of trigger
+type TriggerType string
+
+// A Trigger determines under what circumstances an `event.ETWorkflowTrigger`
+// should be emitted on the given event.Bus. It knows how to `Advance` itself.
+type Trigger interface {
+	// json.Marshaler
+	// json.Unmarshaler
+	ID() ID
```

Comment:
is it weird to have the Trigger and Workflow (and Hook) take the same kind of `ID`? Should I be defining `TriggerID` and `HookID` as types as well?

Reply:
nah, I think it's fine for them to all just use `ID` types